src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_copy_b_emitter.cpp

```diff
@@ -81,31 +82,41 @@ void jit_brgemm_copy_b_emitter::emit_impl(const std::vector<size_t>& in, const s
     if (out.size() > 1)
         mem_ptrs_idxs.emplace_back(out[1]);
 
+    std::set<snippets::Reg> regs_to_spill = m_live_regs;
+    // Note: these 3 registers will be corrupted by the caller during the ABI call
+    regs_to_spill.emplace(snippets::RegType::gpr, abi_param1.getIdx());
+    regs_to_spill.emplace(snippets::RegType::gpr, abi_param2.getIdx());
+    regs_to_spill.emplace(snippets::RegType::gpr, h->rbp.getIdx());
+    // Note: abi_param_1 is a default invalid value to check later that the aux reg was allocated properly
+    Xbyak::Reg64 aux_reg = abi_param1;
+    const bool is_dynamic_case =
+        std::any_of(m_memory_offsets.cbegin(), m_memory_offsets.cend(), ov::snippets::utils::is_dynamic_value<size_t>);
```

Comment:
This code is duplicated in brgemm_emitter.cpp. Moreover, there is a high similarity between these 2 files till L119. Maybe we can extract some common part to utils?

Reply:
The question is still actual, please take a look

Reply:
I applied the changes, but didn't respond to the comment, sorry.
Unfortunately they are not that similar because brgemm_emitter has additional `std::is_same<T, BrgemmAMXKernelExecutor>()` condition. Anyway, I introduced a new helper `init_memory_access_aux_gpr`, it's just a few lines, but better than nothing. Please take a look.

Reply:
UPD introduced a base class to reuse this register allocation code. please take a look